sdk/identity/identity/src/client/sniIdentityClient.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IdentityClient } from "./identityClient.js";
+import { TokenCredentialOptions } from "../tokenCredentialOptions.js";
+import { NetworkRequestOptions, NetworkResponse } from "@azure/msal-node";
+import { createHttpHeaders, createPipelineRequest } from "@azure/core-rest-pipeline";
+import { promises as fs } from 'fs';
+import * as forge from "node-forge";
+
+export class sniIdentityClient extends IdentityClient {
+  private caFileorData: "file" | "data";
+  private caContent: string; // This can be a file path or a string containing the CA certificate data
+  constructor(caFileorData: "file" | "data", caContent: string, options?: TokenCredentialOptions) {
+    super(options);
+    this.caFileorData = caFileorData;
+    this.caContent = caContent;
+  }
+  async sendGetRequestAsync<T>(
+    url: string,
+    options?: NetworkRequestOptions,
+  ): Promise<NetworkResponse<T>> {
+    if (url.includes("/token")) {
```

Comment:
I think this should be `!url.includes("/token")`? We do the new thing only for token requests, and use the old path for everything else.

Reply:
ooh got it!